src/Orckestra.Composer.Cart/Factory/Order/OrderHistoryViewModelFactory.cs

```diff
@@ -235,8 +236,15 @@ protected virtual IEnumerable<OrderHistoryPageViewModel> GetPages(OrderHistoryPa
             orderInfo.OrderStatus = GetOrderStatusDisplayName(rawOrder, param);
             orderInfo.OrderStatusRaw = rawOrder.OrderStatus;
             orderInfo.IsOrderEditable = param.OrderEditingInfos != null  && param.OrderEditingInfos.ContainsKey(Guid.Parse(rawOrder.Id)) && param.OrderEditingInfos[Guid.Parse(rawOrder.Id)];
-            orderInfo.IsOrderCancelable = param.OrderCancellationStatusInfos != null && param.OrderCancellationStatusInfos.ContainsKey(Guid.Parse(rawOrder.Id)) && param.OrderCancellationStatusInfos[Guid.Parse(rawOrder.Id)].CanCancel;
-            orderInfo.IsOrderPendingCancellation = param.OrderCancellationStatusInfos != null && param.OrderCancellationStatusInfos.ContainsKey(Guid.Parse(rawOrder.Id)) && param.OrderCancellationStatusInfos[Guid.Parse(rawOrder.Id)].CancellationPending;
+
+            CancellationStatus orderCancellationStatusInfo = null;
+            if (param.OrderCancellationStatusInfos != null && param.OrderCancellationStatusInfos.ContainsKey(Guid.Parse(rawOrder.Id)))
+            {
+                orderCancellationStatusInfo = param.OrderCancellationStatusInfos[Guid.Parse(rawOrder.Id)];
+            }
+
+            orderInfo.IsOrderCancelable = orderCancellationStatusInfo != null && orderCancellationStatusInfo.CanCancel;
```

Comment:
can be simplified 
orderCancellationStatusInfo?.CanCancel ?? false;
orderCancellationStatusInfo?.CancellationPending?? false;